Sources/TuistKit/Loader/Graph.swift

```diff
@@ -201,3 +254,27 @@ class Graph: Graphing {
         return targetNodes[name]
     }
 }
+
+public struct Stack<T> {
```

Comment:
I'd extract this into its own file. Maybe in [this directory](https://github.com/tuist/tuist/tree/master/Sources/TuistKit/Utils)

Reply:
I'd also add a few unit tests.

Reply:
Moved the code to `Utils/`